packages/perseus-score/src/validate.test.ts

```diff
@@ -0,0 +1,286 @@
+import {
+    getExpressionWidget,
+    getLegacyExpressionWidget,
+    getTestDropdownWidget,
+} from "../util/test-helpers";
+
+import {emptyWidgetsFunctional} from "./validate";
+
+import type {UserInputMap} from "./validation.types";
+import type {PerseusWidgetsMap} from "@khanacademy/perseus-core";
+
+describe("emptyWidgetsFunctional", () => {
+    it("returns an empty array if there are no widgets", () => {
+        // Arrange / Act
+        const result = emptyWidgetsFunctional({}, [], {}, "en");
+
+        // Assert
+        expect(result).toEqual([]);
+    });
+
+    it("properly identifies empty widgets", () => {
+        // Arrange
+        const widgets: PerseusWidgetsMap = {
+            "dropdown 1": getTestDropdownWidget(),
+        };
+        const widgetIds: Array<string> = ["dropdown 1"];
+        const userInputMap: UserInputMap = {
+            "dropdown 1": {
+                value: 0,
+            },
+        };
+
+        // Act
+        const result = emptyWidgetsFunctional(
```

Comment:
There were all changed to no longer require `strings` since we moved to error codes.